connaisseur/alert.py

```diff
@@ -48,6 +48,166 @@ def alerting_required(self, event_category: str) -> bool:
+        """
+        token: str
+        authorization_prefix: str = "Bearer"  # default is bearer
+
+        def __init__(self, alert_receiver_config: dict):
+            bearer_authentication_config = alert_receiver_config.get(
+                "receiver_authentication_bearer", None
+            )
+
+            if (
+                bearer_authentication_config is None
+            ):  # TODO maybe remove this check since it is included in the json validation?
+                raise ConfigurationError(
+                    "No bearer authentication configuration found."
+                )
+
+            token_env = bearer_authentication_config.get("token_env", None)
+            token_file = bearer_authentication_config.get("token_file", None)
+
+            if (
+                token_env is None and token_file is None
+            ):  # TODO maybe remove this check since it is included in the json validation?
+                raise ConfigurationError(
+                    "No token_env and token_file configuration found."
+                )
+
+            if (
+                token_env is not None and token_file is not None
+            ):  # TODO maybe remove this check since it is included in the json validation?
+                raise ConfigurationError(
+                    "Both token_env and token_file configuration found. Only one is required."
+                )
+
+            if token_env is not None:
+                self.token = os.environ.get(token_env, None)
+
+                if self.token is None:
+                    raise ConfigurationError(
+                        f"No token found from environmental variable {token_env}."
+                    )
+            else:
+                try:
+                    with open(token_file, "r") as token_file:
+                        self.token = token_file.read()
+                except FileNotFoundError:
+                    raise ConfigurationError(f"No token file found at {token_file}.")
+                except Exception as err:
+                    raise ConfigurationError(
+                        f"An error occurred while loading the token file {token_file}: {str(err)}"
+                    )
+
+            self.authorization_prefix = bearer_authentication_config.get(
+                "authorization_prefix", "Bearer"
+            )
+            # TODO maybe validate authorization prefix
+
+        def get_header(self) -> dict:
+            return {"Authorization": f"{self.authorization_prefix} {self.token}"}
+
+    def __init__(self, alert_receiver_config: dict):
+        self.authentication_type = alert_receiver_config.get(
+            "receiver_authentication_type", "none"
+        )
+
+        if self.is_basic():
+            self.__init_basic_authentication(alert_receiver_config)
+        elif self.is_bearer():
+            self.__init_bearer_authentication(alert_receiver_config)
```

Comment:
```python
init_map = {
    "basic": AlertReceiverAuthentication.AlertReceiverBasicAuthentication,
    "bearer": AlertReceiverAuthentication.AlertReceiverBearerAuthentication,
}
try:
    self.auth_instance = init_map[self.authentication_type](alert_receiver_config)
except AttributeError:
    ....
```

Then, it `get_auth_header()`, you don't need to do the case handling, but can just return `{} if self.auth_instance is None self.auth_instance.get_header()`.
```